File: Python/Programmars/Level_2/위장.py
# ref: https://eda-ai-lab.tistory.com/472
def solution(clothes):
    answer = {}
    for i in clothes:
        answer[i[1]] = answer.get(i[1], 0) + 1
    count = 1
    for i in answer.values():
        count *= (i+1)  # 안 입은 경우를 고려하여 +1
    return count - 1  # 모두 안 입은 경우 제외


print(solution([["yellowhat", "headgear"], [
      "bluesunglasses", "eyewear"], ["green_turban", "headgear"]]))  # 5
print(solution([["crowmask", "face"], [
      "bluesunglasses", "face"], ["smoky_makeup", "face"]]))  # 3

File: JavaScript/Programmars/해시/위장.js
function solution(clothes) {
  const clothesMap = new Map();
  for (let [, type] of clothes) {
    clothesMap.set(type, (clothesMap.get(type) || 0) + 1);
  }

  let answer = 1;
  for (let val of clothesMap.values()) {
    answer *= val + 1; // 종류 수 + 착용 X
  }

  return answer - 1; // 아무것도 착용하지 않는 경우 제외
}

console.log(
  solution([
    ['yellowhat', 'headgear'],
    ['bluesunglasses', 'eyewear'],
    ['green_turban', 'headgear'],
  ])
); // 5
console.log(
  solution([
    ['crowmask', 'face'],
    ['bluesunglasses', 'face'],
    ['smoky_makeup', 'face'],
  ])
); // 3
